web/scripts/main.js

```diff
@@ -253,10 +261,29 @@ class Design {
                 case "raster": {
                     const bm = await layer.get_raster_bitmap();
                     const imgdata = await yak.ImageData_from_ImageBitmap(bm);
-                    gingerbread.conversion_add_raster_layer(layer.number, this.trace_scale_factor, imgdata);
+
+                    // Check that the ImageData is valid
+                    const imgdata_sum = imgdata.data.reduce((a, b) => a + b, 0);
```

Comment:
This could be `0` for an all black image, which would still be a valid, if silly, image. Would it be better to check if the length is 0?